ui/src/app/views/application/show/workflow/application.workflow.component.ts

```diff
@@ -193,29 +191,39 @@ export class ApplicationWorkflowComponent implements OnInit {
      * Action when changing branch
      */
     changeBranch(): void {
-        // reinit verison filter
-        this.applicationFilter.version = '';
-        jQuery('.cdsVersion div.text')[0].textContent = '';
-        this.changeVersion();
-
         // Load the versions of the new branch
-        this.loadVersions(this.project.key, this.application.name);
+        this.loadVersions(this.project.key, this.application.name)
+            .subscribe(() => this.changeVersion());
     };
 
     /**
      * Action when changing version
      */
-    changeVersion(): void {
+    changeVersion(version?: number): void {
         this.applicationFilter.branch = this.applicationFilter.branch.trim();
-        if (this.applicationFilter.version.trim() === '') {
-            this.applicationFilter.version = 0;
+
+        if (!version && Array.isArray(this.versions) && this.versions.length) {
+            this.applicationFilter.version = this.versions[1];
+        }
+
+        if (version) {
+            this.applicationFilter.version = version;
         }
+
         this._router.navigate(['/project/', this.project.key, 'application', this.application.name],
             {queryParams: {tab: 'workflow', branch: this.applicationFilter.branch, version: this.applicationFilter.version}});
         this.changeWorkerEvent.emit(true);
         this.clearTree(this.application.workflows);
     }
 
+    /**
+     * Load the list of version for the current application on the selected branch
+     */
+    loadVersions(key: string, appName: string): Observable<Array<number>> {
+        return this._appWorkflow.getVersions(key, appName, this.applicationFilter.branch)
+            .map((versions) => this.versions = versions);
+    };
+
```

Comment:
Yes but when you remove the filter it still takes the default branch in the filter. So there is always a filter if I can say that. Moreover, on the first visit when you don't choose filter branch or version it display an empty filter in the ui. I think it's better to not have multiple types in an array (string and number). And with this modification we don't have to override the dom with Jquery and it's better for angular bindings.